packages/polar/src/lib/deploy/contract.ts

```diff
@@ -9,60 +9,173 @@ import {
+    argNames.forEach((abiParam, i) => {
+      msgArgs[abiParam.name] = args[i];
+    });
+
+    // Query the current count
+    return contract.executeMsg(msgName, msgArgs, account);
+  };
+}
 
 export class Contract {
   readonly contractName: string;
   readonly contractPath: string;
-  readonly schemaPath: string;
+  readonly initSchemaPath: string;
+  readonly querySchemaPath: string;
+  readonly executeSchemaPath: string;
+  readonly initAbi: Abi;
+  readonly queryAbi: Abi;
+  readonly executeAbi: Abi;
   readonly env: PolarRuntimeEnvironment;
   readonly client: CosmWasmClient;
 
   private codeId: number;
-  private readonly contractCodeHash: string;
+  private contractCodeHash: string;
   private contractAddress: string;
 
+  public query: {
+    [name: string]: ContractFunction<any>
+  };
+
+  public tx: {
+    [name: string]: ContractFunction<any>
+  };
+
   constructor (contractName: string, env: PolarRuntimeEnvironment) {
-    this.contractName = contractName;
+    this.contractName = contractName.replace('-', '_');
     this.codeId = 0;
     this.contractCodeHash = "mock_hash";
-    this.contractAddress = "mock_address";
-    this.contractPath = path.join(ARTIFACTS_DIR, "contracts", `${contractName}.wasm`);
-    this.schemaPath = path.join(SCHEMA_DIR, `${contractName}.json`);
+    this.contractAddress = "secret15l3z73lvy0357qrnyhcfpcxhdrxc209wpvkayl";
+    this.contractPath = path.join(ARTIFACTS_DIR, "contracts", `${this.contractName}_compressed.wasm`);
+
+    this.initSchemaPath = path.join(SCHEMA_DIR, this.contractName, "init_msg.json");
+    this.querySchemaPath = path.join(SCHEMA_DIR, this.contractName, "query_msg.json");
+    this.executeSchemaPath = path.join(SCHEMA_DIR, this.contractName, "handle_msg.json");
+
+    if (!fs.existsSync(this.initSchemaPath)) {
+      console.log("Warning: Init schema not found for contract ", chalk.cyan(contractName));
+    }
+    if (!fs.existsSync(this.querySchemaPath)) {
+      console.log("Warning: Query schema not found for contract ", chalk.cyan(contractName));
+    }
+    if (!fs.existsSync(this.executeSchemaPath)) {
+      console.log("Warning: Execute schema not found for contract ", chalk.cyan(contractName));
+    }
+
+    const initSchemaJson: AnyJson = fs.readJsonSync(this.initSchemaPath);
+    const querySchemaJson: AnyJson = fs.readJsonSync(this.querySchemaPath);
+    const executeSchemaJson: AnyJson = fs.readJsonSync(this.executeSchemaPath);
+    this.initAbi = new Abi(initSchemaJson);
+    this.queryAbi = new Abi(querySchemaJson);
+    this.executeAbi = new Abi(executeSchemaJson);
+
+    this.query = {};
+    this.tx = {};
 
     this.env = env;
     this.client = getClient(env.network);
+  }
+
+  async parseSchema (): Promise<void> {
+    await this.queryAbi.parseSchema();
+    await this.executeAbi.parseSchema();
 
-    if (!fs.existsSync(this.contractPath)) {
-      throw new PolarError(ERRORS.ARTIFACTS.NOT_FOUND, {
-        param: this.contractName
-      });
```

Comment:
where are we checking if `user hasn't compiled contract we will compile it with a warning` ?

Reply:
In compress function